doc/design/web.md

```diff
@@ -0,0 +1,153 @@
+
+One example is [AWS IAM](https://aws.amazon.com/iam/?nc2=h_m1), but we can do more simpler than that.
+
+The current implementation under this repo can only have one user for one Kubernetes namespace. We can implement multi-tenant in the near future.
+
+Once a user logged in, s/he will be redirected to the "Job Dashboard" page.
+
+## Jupiter Notebook
+
+Start a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) using image `docker.paddlepaddle.org/book` in the Kubernetes cluster and add an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) endpoint when a user first enters the notebook page. This is already implemented.
+
+<img src="pictures/notebook.png" width="500px" align="center">
+
+Users can write a program in python in the web page and save their programs, which will be saved at cloud storage. Users also can run a script like below to submit a cluster training job:
+
+```python
+create_cloud_job(
+    name,
+    num_trainer,
+    mem_per_trainer,
+    gpu_per_trainer,
+    cpu_per_trainer,
+    num_ps,
+    mem_per_ps,
+    cpu_per_ps,
+)
+```
+
+After this, there will be a job description and performance monitoring pages able to view at "Job Dashboard"
+
+## Job Dashboard
+
+### Job history and currently running jobs
+
+A web page containing a table to list jobs satisfying user's filters. The user can only list jobs that were submitted by themselves.
+
+| jobname | start time | age      | success | fails | actions |
+| ------- | ---------- | -------- | ------- | ----- | ------- |
+| test1   | 2017-01-01 | 17m      |    0    |   0   | stop/log/perf |
+
+Users can filter the list using:
+
+- status:  running/stopped/failed
+- time:    job start time
+- jobname: search by jobname
+
+Viewing job logs:
+
+Click the "log" button on the right of the job will pop up a console frame at the bottom of the page showing the tail of the job log, here shows the first pod's log. On the left side of pop up console frame, there should be a vertical list containing the list of pods, then click one of the pod, the console will show it's log.
+
+### Performance Monitoring
+
+A web page containing graphs monitoring job's resource usages according to time change:
+
+- CPU usage
+- GPU usage
+- memory usage
+- network bandwidth
+- disk I/O
+
+### Quota Monitoring
+
+A web page displaying total quota and quota currently in use.
+
+Also display total CPU time, GPU time in latest 1day, 1week, and 1month.
+
+## Datasets and Models
+
+Datasets and Models are quite the same, both like a simple file management and sharing service.
```

Comment:
Maybe we can supplement more information about the file sharing service.Such as we can share files between users, namespaces or just a publish link?